### PR TITLE
Json models

### LINKED ISSUE #72: Change local https status to http.HTTPStatus
I didn't notice that HTTPStatus is a python standard library

### LINKED ISSUE #74: Add json methods to db model classes
It'd be a good idea to get the GET replies from a model method. Look into jsonify.

### LINKED ISSUE #75: Nuke Migration
The migration is really broken. We likely need to nuke the migration and restart. Check that it hasn't reversed `upgrade`/`downgrade` in the new version.

### PR DESCRIPTION
**Pull request by giancarlopernudisegura**
_I'm also nuking the migrations since they were messed up giving us a clean slate._

Create JSON methods for the DB models so they can be called from different APIs reducing duplicate code. I've created and modified some of the API endpoints to showcase what this would look like, including ones with pagination. Finally, I've removed the API prefix to better match the specifications. I've added some docstrings and typing hints to make things easier to follow and use.

This should provide a good showcase of what to do for the rest of the project.

You should delete your database, and run `flask db upgrade` before testing out this branch.

### CODE REVIEW

**Comment by Pegmode** [comment]
Resolves https://github.com/giancarlopernudisegura/cmput404/issues/75 and https://github.com/giancarlopernudisegura/cmput404/issues/74

**Comment by Pegmode** [comment]
Trying to start npm:
```[webpack-cli] Failed to load '/home/dan/cmput404/webpack.dev.js' config
[webpack-cli] Error: Cannot find module 'dotenv'
Require stack:
- /home/dan/cmput404/webpack.dev.js
- /home/dan/cmput404/node_modules/webpack-cli/lib/webpack-cli.js
- /home/dan/cmput404/node_modules/webpack-cli/lib/bootstrap.js
- /home/dan/cmput404/node_modules/webpack-cli/bin/cli.js
- /home/dan/cmput404/node_modules/webpack-dev-server/bin/webpack-dev-server.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/dan/cmput404/webpack.dev.js:3:14)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/dan/cmput404/webpack.dev.js',
    '/home/dan/cmput404/node_modules/webpack-cli/lib/webpack-cli.js',
    '/home/dan/cmput404/node_modules/webpack-cli/lib/bootstrap.js',
    '/home/dan/cmput404/node_modules/webpack-cli/bin/cli.js',
    '/home/dan/cmput404/node_modules/webpack-dev-server/bin/webpack-dev-server.js'
  ]
}

**Comment by Pegmode** [comment]
`flask run` 
```
Error: While importing 'server.run', an ImportError was raised.
```

`python3 run.py`
```Traceback (most recent call last):
  File "run.py", line 4, in <module>
    from server.routes import api
ModuleNotFoundError: No module named 'server'

**Comment by giancarlopernudisegura** [comment]
For the npm error, did you run `npm install` beforehand? I didn't mess with the package.json in this PR so it shouldn't be an issue.

**Comment by Pegmode** [comment]
Fixed `npm` issue by running `npm install`, fixed flask by nuking and re-creating local virtual-env

**Comment by Pegmode** [comment]
Tested all the API endpoints except for login and all the available http methods on them. They all function as expected. 

It was already said but I'd like to reiterate that the database needs to be purged and the migration needs to be run otherwise there are a whole bunch of foreign key errors. 

Also for testing authors, make sure there is a valid github id otherwise there are a bunch of HTTP errors from the author model

Tests:
- GET `/authors`
- GET `/authors/<int:author_id>`
- POST `/authors/<int:author_id>`  (not implemented but works)
- GET `/authors/<int:author_id>/posts/<int:post_id>`
- GET `/authors/<int:author_id>/posts/`
- POST `/authors/<int:author_id>/posts/`

File: server/models.py [hunk 1]
```diff
@@ -13,33 +23,44 @@ class Author(db.Model):
...
-        self.isVerified = isVerified#should set this from a JSON or db later
-
+        self.isVerified = isVerified  # should set this from a JSON or db later
 
     def __repr__(self):
         return f"<id {self.id}>"
 
+    def json(self) -> Dict[str, str]:
+        # get username from github id
+        resp = urlopen(f'https://api.github.com/user/{self.githubId}')
```
Comment (Pegmode) [inline server/models.py:38]: If the github ID is invalid this raises a 404 ERROR exception. For our side we should expect that unless we are manually creating a debug or testing author this wont be an issue but I wonder about when we have to hook our API to other groups. If we fix this I suggest only calling `author.json()` in a try except block.

Comment (giancarlopernudisegura) [reply server/models.py:38]: Yeah, when we hook it up to other groups then we'll have to look at making this more resilient but for now, I think it's ok.

**Review by Pegmode — APPROVED** [review]
Looks good. /login API endpoint was not able to be tested because firebase is throwing errors. That is not the responsibility of this branch so I am comfortable with merging this

**Pull request merged**